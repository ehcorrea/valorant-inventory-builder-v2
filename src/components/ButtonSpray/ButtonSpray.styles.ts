import { TouchableOpacity } from 'react-native';
import styled, { css } from 'styled-components/native';
import { Image, ImageProps } from 'expo-image';

export const Button = styled(TouchableOpacity)`
  ${({ theme }) => css`
    align-items: center;
    aspect-ratio: 1/1;
    background-color: #cfcfcf11;
    border-radius: ${theme.border.radius.medium}px;
    flex: 1;
    justify-content: center;
    max-height: 250px;
    max-width: 250px;
  `}
`;

export const ImageSpray = styled(Image).attrs<ImageProps>({
  contentFit: 'cover',
})`
  height: 90%;
  width: 90%;
`;
