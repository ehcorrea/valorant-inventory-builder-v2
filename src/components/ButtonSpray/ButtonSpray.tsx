import { TouchableOpacityProps } from 'react-native';

import * as S from './ButtonSpray.styles';

export type ButtonSprayProps = {
  image: string;
} & TouchableOpacityProps;

export function ButtonSpray({ image, ...props }: ButtonSprayProps) {
  return (
    <S.Button {...props}>
      <S.ImageSpray source={image} />
    </S.Button>
  );
}
