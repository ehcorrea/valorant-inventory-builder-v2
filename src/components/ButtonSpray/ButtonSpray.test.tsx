import { fireEvent, render } from '@/test/utils';

import { ButtonSpray } from './ButtonSpray';

const mockedOnPress = jest.fn();

const setup = () => {
  return render(
    <ButtonSpray
      onPress={mockedOnPress}
      image="placeholder-image.png"
      testID="weapon-button"
    />
  );
};

describe('given that the component is rendered', () => {
  test('render image passed by props', () => {
    const container = setup();
    expect(
      container
        .getByTestId('weapon-button')
        .findByProps({ source: 'placeholder-image.png' })
    ).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('pressing calls a callback', () => {
    const container = setup();
    const button = container.getByTestId('weapon-button');
    fireEvent.press(button);
    expect(mockedOnPress).toHaveBeenCalled();
  });
});
