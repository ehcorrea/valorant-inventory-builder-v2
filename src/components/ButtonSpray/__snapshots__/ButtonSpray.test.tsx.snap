// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given that the component is rendered render image passed by props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "aspectRatio": 1,
      "backgroundColor": "#cfcfcf11",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "maxHeight": 250,
      "maxWidth": 250,
      "opacity": 1,
    }
  }
  testID="weapon-button"
>
  <Image
    contentFit="cover"
    source="placeholder-image.png"
    style={
      {
        "height": "90%",
        "width": "90%",
      }
    }
  />
</View>
`;
